octopus_deploy/tests/test_unit.py

```diff
@@ -13,7 +13,7 @@
 from datadog_checks.dev.utils import get_metadata_metrics
 from datadog_checks.octopus_deploy import OctopusDeployCheck
 
-from .constants import ALL_METRICS
+from .constants import ALL_DEPLOYMENT_LOGS, ALL_METRICS, ONLY_TEST_LOGS
 
 MOCKED_TIME1 = datetime.datetime.fromisoformat("2024-09-23T14:45:00.123+00:00")
```

Comment:
```suggestion
MOCKED_TIME1 = datetime.datetime.fromisoformat("2024-09-23T14:45:00.123+00:00")
```
I'm not sure what's the idea behind `MOCKED_TIME1` ?
And how does this timestamp related to the timestamps from `ALL_DEPLOYMENT_LOGS`?

Reply:
`MOCKED_TIME1` and `MOCKED_TIME2` determine what the `_from_completed_time` and `_to_completed_time` variables are for each check run. We updated the times every check run: https://github.com/DataDog/integrations-core/blob/sarah/add-octopus-integration/octopus_deploy/datadog_checks/octopus_deploy/check.py#L39-L44, making sure the `_from_completed_time` is always 15 seconds behind `_to_completed_time` (except the first check run). We use these to ensure we only get completed tasks that we haven't seen before: https://github.com/DataDog/integrations-core/blob/sarah/add-octopus-integration/octopus_deploy/datadog_checks/octopus_deploy/check.py#L221-L229.

Reply:
I see now, makes sense! This is also the case for the logs collection right? Making sure we don't miss any logs between 2 check runs?

On a second note, I noticed this in the unit tests, it might be good to assert that the length of the metrics returned on this line is not empty or something, unless we're okay with not asserting anything if it's empty:
https://github.com/DataDog/integrations-core/blob/bb3bb642abe67b875255f9f97984307ebe274028/octopus_deploy/tests/test_unit.py#L312-L313

Reply:
yes correct! good idea, i added an extra assertion!